test/origin_simulator/origin_simulator_test.exs

```diff
@@ -91,162 +96,143 @@ defmodule OriginSimulatorTest do
 
-  describe "GET page with origin" do
+  describe "GET page" do
     test "will return the origin page" do
       payload = %{
         "origin" => "https://www.bbc.co.uk/news",
-        "stages" => [%{ "at" => 0, "status" => 200, "latency" => 0}]
+        "stages" => [%{"at" => 0, "status" => 200, "latency" => 0}]
       }
 
       conn(:post, "/add_recipe", Poison.encode!(payload)) |> OriginSimulator.call([])
-
-      Process.sleep 20
+      Process.sleep(20)
 
       conn = conn(:get, "/")
       conn = OriginSimulator.call(conn, [])
 
       assert conn.state == :sent
       assert conn.status == 200
       assert get_resp_header(conn, "content-type") == ["text/html; charset=utf-8"]
-      assert conn.resp_body == "some content from origin"
+      assert conn.resp_body == body_mock()
     end
-  end
 
-  describe "GET page with content" do
-    test "will return the passed content" do
+    test "will return the origin page with random latency within range" do
       payload = %{
-        "body" =>   "{\"hello\":\"world\"}",
-        "stages" => [%{ "at" => 0, "status" => 200, "latency" => 0}]
+        "origin" => "https://www.bbc.co.uk/news",
+        "stages" => [%{"at" => 0, "status" => 200, "latency" => "10ms..50ms"}]
       }
 
       conn(:post, "/add_recipe", Poison.encode!(payload)) |> OriginSimulator.call([])
-
-      Process.sleep 20
+      Process.sleep(20)
 
       conn = conn(:get, "/")
       conn = OriginSimulator.call(conn, [])
 
       assert conn.state == :sent
       assert conn.status == 200
-      assert get_resp_header(conn, "content-type") == ["application/json; charset=utf-8"]
-      assert conn.resp_body == "{\"hello\":\"world\"}"
+      assert get_resp_header(conn, "content-type") == ["text/html; charset=utf-8"]
+      assert conn.resp_body == body_mock()
     end
-  end
 
-  describe "GET page with random content" do
-    test "will return random content of the passed size" do
+    test "will return the passed content" do
       payload = %{
-        "random_content" =>   "50kb",
-        "stages" => [%{ "at" => 0, "status" => 200, "latency" => 0}]
+        "body" => "{\"hello\":\"world\"}",
+        "stages" => [%{"at" => 0, "status" => 200, "latency" => 0}]
       }
 
       conn(:post, "/add_recipe", Poison.encode!(payload)) |> OriginSimulator.call([])
-
-      Process.sleep 20
+      Process.sleep(20)
 
       conn = conn(:get, "/")
       conn = OriginSimulator.call(conn, [])
 
       assert conn.state == :sent
       assert conn.status == 200
-      assert get_resp_header(conn, "content-type") == ["text/html; charset=utf-8"]
-      assert String.length(conn.resp_body) == 50 * 1024
+      assert get_resp_header(conn, "content-type") == ["application/json; charset=utf-8"]
+      assert conn.resp_body == body_mock(type: :json)
     end
-  end
 
-  describe "GET page with random latency within range" do
-    test "will return the origin page" do
+    test "will return the passed content with respond headers" do
```

Comment:
Probably a typo - I assume it's meant to be response headers.

Secondly, parsed content or passed as in passing through?

Reply:
Typo yeah (response). I also wondered about `passed` vs. `parsed`. This term was already there perhaps to refer to parsing the encoded JSON of the `body` recipe field. So probably `parsed`. I think we should just use `body` to be consistent.

Reply:
Using `parsed body content` now.